capsul/engine/module/spm.py

```diff
@@ -170,9 +171,11 @@ def check_notably_invalid_config(conf):
         list of invalid config keys
     '''
     invalid = []
-    for k in ('directory', 'version'):
-        if getattr(conf, k, None) is None:
-            invalid.append(k)
+    #for k in ('directory', 'version'):
+    #    if getattr(conf, k, None) is None:
+    #        invalid.append(k)
+    if getattr(conf, 'directory', None) is None:
```

Comment:
Why did "version" become optional ? In some processes we need to know which version of SPM we are using (even if now we are always using SPM12)

Reply:
Simply because currently, if the version is not defined, the process runs anyway, so if the test declares the config invalid, we find ourselves in a situation where the process is declared invalid while the process can run ... this is based just on an observation , maybe we need to change the machinery about this ... ???

Reply:
Currently (at least) the version is part of the config-id key of SPM modules, so it is mandatory. Well, I don't know what happens if we don't specify it, actually.

Reply:
@denisri you approved the change but do you want us to investigate further to improve the machinery for the version  config-id key od SPM or can we merge with master?

Reply:
tests are failing, is it linked with the changes ?

Reply:
I just did a roll back for engine.module.spm.config_dependencies() (for the `version` I don't see a problem with the change).
I won't go into the details of why I had made the change but I think there is a leak for matlab MCR (we should talk about this as it could take a long time to write).
Unit tests now work on the capsul side and on the populse_mia side we now do what we wanted (test the configuration before running a process). This is only valid for a local operation ...
I think this PR can be merged now...